lib/rust/parser/debug/tests/parse.rs

```diff
@@ -1189,6 +1189,35 @@ fn pattern_match_auto_scope() {
     test(&code.join("\n"), expected);
 }
 
+// === Private (project-private) keyword ===
+#[test]
+fn private_keyword() {
+    test("private", block![(Private)]);
+}
+
+#[test]
+#[ignore]
+fn private_is_first_statement() {
```

Comment:
This will require changes to the macro resolver--we could do something like:
`enum Context {
    Expression,
    Statement { first: bool }
}`
The resolver would then keep track of when the first "contentful' line has been encountered in order to construct the `first` field. Then we could add a third `MacroMap` that is only run in `Statement` context with `first=true`. Note that there's a rewrite of the macro resolver currently in CI: #7711.